datasets/glossis/interpolate_vtk.py

```diff
@@ -0,0 +1,248 @@
+    return polydata, mask 
+
+def update_data(path, polydata, mask, variables, t=0):
+    """ Add arrays for each variable to polydata """
+    ds = netCDF4.Dataset(path)
+    for key in variables:
+        arr = ds.variables[key][t, ...]
+
+        array = tvtk.FloatArray()
+        array.from_array(arr[~mask])
+        array.name = key
+        polydata.cell_data.add_array(array)
+    polydata.modified()
+    ds.close()
+
+def setup_pipeline(polydata, width=1000, height=1000):
+    """ VTk pipeline to resample data """
+    resample = tvtk.ResampleToImage()
+    cell2point = tvtk.CellDataToPointData()
+    # setup the pipeline
+    configure_input(cell2point, polydata)
+
+    configure_input(resample, cell2point)
+
+    resample.sampling_dimensions = np.array([width, height, 1])
+    resample.sampling_bounds = np.array([-180, 180, -90, 90, 0, 0.0])
+    resample.use_input_bounds = False
+    # TODO: check if this can be left out
+    resample.modified()
+    resample.update()
+    polydata.modified()
+
+    return resample
+
+def resample_data(netcdfs, img_size, t, variables):
+    """ Resample data for timestep, variables in netcdf to given image size """
+    polydatas_by_name = {}
+
+    mbd = tvtk.MultiBlockDataSet()
+    mbd.number_of_blocks = len(netcdfs)
+
+    masks_by_name = {}
+
+    for i, f in enumerate(netcdfs):
+        ds = netCDF4.Dataset(f)
+        grid = gridded.pyugrid.ugrid.UGrid.from_nc_dataset(ds)
+        polydata, mask =  grid2poly(grid)
+        mbd.set_block(i, polydata)
+        masks_by_name[f.name] = mask
+        polydatas_by_name[f.name] = polydata
+        ds.close()
+
+    mbm = tvtk.MultiBlockMergeFilter()
+    configure_input_data(mbm, mbd)
+    resample = setup_pipeline(mbm, width=img_size[0], height=img_size[1])
+
+    for path in netcdfs:
+        polydata = polydatas_by_name[path.name]
+        mask = masks_by_name[path.name]
+        update_data(path, polydata, mask, variables, t=0)
+    resample.update()
+    mbm.update()
+
+    return resample
+
+def fm_to_tiff_vtk(
+    bucketname,
+    prefixname,
+    tmpdir,
+    variables=["currents_u", "currents_v"],
+    filter="currents",
+    output_fn="glossis_currents",
+    nodata=-9999,
+    extra_bands=0,
+):
+    """Convert FM netcdfs in bucket into geotiffs for each timestep using vtk."""
+
+    # Download netcdfs files from bucket
+    netcdfs = download_netcdfs_from_bucket(
+        bucketname, prefixname, tmpdir, filter)
+    netcdfs = [pathlib.Path(netcdf) for netcdf in netcdfs]
+
+    path = pathlib.Path('tmp/netcdfs')
+    netcdfs = list(path.glob('*.nc'))
+    
+    if len(netcdfs) == 0:
+        logger.warning(f"No {filter} files found!")
+        return []
+
+    # Determine timesteps in first netcdf
+    nc = netCDF4.Dataset(netcdfs[0])
+    timesteps = netCDF4.num2date(
+        nc.variables["time"][:], units=nc.variables["time"].units
+    )
+    logger.info(
+        "{} timesteps of which only the first six will be processed.".format(
```

Comment:
Don't forget to update this comment if this changes.

Reply:
Yes will do